scripts/dts/gen_defines.py

```diff
@@ -463,6 +471,8 @@ def encode_zephyr_multi_level_irq(irq, irq_num):
         for cell_name, cell_value in irq.data.items():
             ident = "IRQ_{}".format(irq_i)
             if cell_name == "irq":
+                if "arm,gic" in irq.controller.compats:
+                    cell_value = map_arm_gic_irq_type(irq, irq.data["type"], cell_value)
```

Comment:
Could check if `"type"` exists in the dictionary and call `err()` otherwise (see elsewhere in the script). Maybe `map_arm_gic_irq_type()` could take just `irq`, to move the check into it.

Probably will never happen, but the goal is for the script to never crash with a generic Python error.

Reply:
Added check for `type` definition as well as its value.

Reply:
Thanks!